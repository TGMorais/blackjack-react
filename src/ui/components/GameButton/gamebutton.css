.gamebutton {
    font-family: 'Dosis', 'sans serif';
    font-weight: bold;
    margin: 0 1rem;
    padding: 1rem;
    border-radius: 1.5rem;
    font-size: 2rem;
    background: transparent;
    color: #ececec;
    border-color: #ff004b;
    box-shadow: 0 19px 38px rgba(0,0,0,0.30), 0 15px 12px rgba(0,0,0,0.22);    
}

.gamebutton:hover {
    cursor: pointer;
    border-color: #ececec;
    transform: scale(1.2);
}
