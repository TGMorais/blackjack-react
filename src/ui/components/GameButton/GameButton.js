import React from "react";
import "./gamebutton.css";

const GameButton = props => (
  <button className={'gamebutton'} {...props}>
    {props.children}
  </button>
);

export default GameButton;
