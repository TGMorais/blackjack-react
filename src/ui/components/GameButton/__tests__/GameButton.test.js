import React from "react";
import ReactDOM from "react-dom";
import GameButton from "../GameButton";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<GameButton />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders correctly", () => {
  const tree = renderer.create(<GameButton>Button</GameButton>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Change state when hovered", () => {
  var clicked = false;
  const component = renderer.create(
    <GameButton onClick={() => {clicked = true}}>Button</GameButton>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  tree.props.onClick();
  tree = component.toJSON();
  expect(clicked).toBe(true)
});
