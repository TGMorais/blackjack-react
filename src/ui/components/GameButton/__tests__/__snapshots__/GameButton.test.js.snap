// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change state when hovered 1`] = `
<button
  className="gamebutton"
  onClick={[Function]}
>
  Button
</button>
`;

exports[`renders correctly 1`] = `
<button
  className="gamebutton"
>
  Button
</button>
`;
